### PR TITLE
feat: add multiple providers in addition to anthropic

### PR DESCRIPTION
**Pull request by rrfaria**
Hi, I would like to share a contribution

this MR allow project to work with multiple providers

now you can use gemini, openAi, ollama, anthropic and if you would like to add more providers you can fallow this docs:
https://sdk.vercel.ai/providers/ai-sdk-providers

### CODE REVIEW

**Comment by rrfaria** [comment]
this is a demo test using gemini api:

image

**Comment by timoa** [comment]
Great contribution! Maybe we can merge your code with the @coleam00 fork?

He had ability to select the LLM provider and model from the landing page and during the chat step:
Screenshot 2024-10-16 at 11 50 37

Repo: https://github.com/coleam00/bolt.new-any-llm

You can see it on his YouTube video:
https://www.youtube.com/watch?v=3PFcAu_oU80

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -60,6 +60,10 @@ pnpm install
 
 ```
 ANTHROPIC_API_KEY=XXX
+PROVIDER= gemini | antrophic | openai | ollama
```
Comment (ijijioritain) [inline CONTRIBUTING.md:63]: typo - did you mean to spell anthropic ? @rrfaria

**Comment by Sunwood-ai-labs** [comment]
I implemented Bedrock based on https://github.com/coleam00/bolt.new-any-llm!  (#689)

**Review by dustinwloring1988 — APPROVED** [review]
Look good, thanks for adding this I will keep using it in my local instance

File: app/lib/.server/llm/providers/ollama.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { createOllama } from 'ollama-ai-provider';
+import type { ModelFactory } from './modelFactory';
+
+export function getOllamaModel(apiKey: string, modelName: string = 'llama3.2:latest') {
+  const model = createOllama({
+    baseURL: 'http://172.21.208.1:11434',
```
Comment (hacknlove) [inline app/lib/.server/llm/providers/ollama.ts:6]: I think this should be in an environmental variable

Comment (rrfaria) [reply app/lib/.server/llm/providers/ollama.ts:6]: I abandoned this PR in favor of [this fork](https://github.com/coleam00/bolt.new-any-llm)
due to inactivity of developers here.

I recommend you to use this fork instead.

**Comment by davych** [comment]
that's so Incredible

**Pull request closed without merging**